src/gui/topbar_subwindows/gui_production_window.hpp

```diff
@@ -333,67 +320,53 @@ class factory_subsidise_button : public button_element_base {
+			} else {
+				if(disabled) {
+					text::add_line(state, contents, "production_not_allowed_to_subsidise_tooltip");
+					text::add_line(state, contents, "cant_subsidize_explanation");
 				} else {
-					if(disabled) {
-						text::add_line(state, contents, "production_not_allowed_to_subsidise_tooltip");
-						text::add_line(state, contents, "cant_subsidize_explanation");
-					} else {
-						text::add_line(state, contents, "production_allowed_to_subsidise_tooltip");
-					}
+					text::add_line(state, contents, "production_allowed_to_subsidise_tooltip");
 				}
-				text::add_line_break_to_layout(state, contents);
-				text::add_line(state, contents, "production_subsidies_desc");
 			}
-
-			// TODO: classic tooltips ???
+			text::add_line_break_to_layout(state, contents);
+			text::add_line(state, contents, "production_subsidies_desc");
 		}
 	}
 };
 
 class factory_delete_button : public button_element_base {
 public:
 	void on_update(sys::state& state) noexcept override {
-		if(parent) {
-			auto fid = retrieve<dcon::factory_id>(state, parent);
-			auto n = retrieve<dcon::nation_id>(state, parent);
-
-			disabled = !command::can_delete_factory(state, state.local_player_nation, fid);
-		}
+		const dcon::factory_id fid = retrieve<dcon::factory_id>(state, parent);
+		const dcon::nation_id n = retrieve<dcon::nation_id>(state, parent);
+		disabled = !command::can_delete_factory(state, n, fid);
 	}
 
 	void button_action(sys::state& state) noexcept override {
-		if(parent) {
-			auto fid = retrieve<dcon::factory_id>(state, parent);
-			auto n = retrieve<dcon::nation_id>(state, parent);
-
-			command::delete_factory(state, state.local_player_nation, fid);
-		}
+		const dcon::factory_id fid = retrieve<dcon::factory_id>(state, parent);
+		const dcon::nation_id n = retrieve<dcon::nation_id>(state, parent);
+		command::delete_factory(state, n, fid);
 	}
 
 	tooltip_behavior has_tooltip(sys::state& state) noexcept override {
 		return tooltip_behavior::variable_tooltip;
 	}
 
 	void update_tooltip(sys::state& state, int32_t x, int32_t y, text::columnar_layout& contents) noexcept override {
-		if(parent) {
-			auto fid = retrieve<dcon::factory_id>(state, parent);
-			auto n = retrieve<dcon::nation_id>(state, parent);
-
-			if(n == state.local_player_nation) {
-				text::add_line(state, contents, "factory_delete_header");
-				if(disabled) {
-					text::add_line_break_to_layout(state, contents);
-					text::add_line(state, contents, "factory_delete_not_allowed");
-				}
-				// TODO: classic tooltips ?
+		const dcon::factory_id fid = retrieve<dcon::factory_id>(state, parent);
+		const dcon::nation_id n = retrieve<dcon::nation_id>(state, parent);
+		if(n == state.local_player_nation) {
+			text::add_line(state, contents, "factory_delete_header");
+			if(disabled) {
+				text::add_line_break_to_layout(state, contents);
+				text::add_line(state, contents, "factory_delete_not_allowed");
 			}
+			// TODO: classic tooltips ?
```

Comment:
I guess these comments can be removed, since the special setting for them is no more